TrackletAlgorithm/MatchEngineUnit.h

```diff
@@ -101,18 +64,142 @@ class MatchEngineUnit : public MatchEngineUnitBase<VMProjType> {
   stubmask_[3] = nstubsall_[3]!=0;
   ap_uint<2> index = __builtin_ctz(stubmask_);
   stubmask_[index]=0;
-  second_ = isSecond[index];
-  phiPlus_ = isPhiPlus[index];
+  second_ = index[0];
+  phiPlus_ = index[1];
   nstubs_ = nstubsall_[index];
-  ivmphi = projbuffer.getPhi();
-  iphi_ = iphi;
-  auto const qdata=projbuffer_;
-  tcid=qdata.getTCID();
+  iphi_ = projbuffer.getPhi();
+  tcid_ = projbuffer.getTCID();
 
-  good_ = false;
+  good__ = false;
 
 }
 
+
+
+inline void step(const VMStubMECM<VMSMEType> stubmem[2][1024]) {
+#pragma HLS inline
+#pragma HLS array_partition variable=nstubsall_ complete dim=1
+
+  bool nearfull = nearFull();
+
+  good__ = (!idle_) && (!nearfull);
+
+  // Buffer still has projections to read out
+  //If the buffer is not empty we have a projection that we need to 
+  //process. 
+  
+  NSTUBS istubtmp=istub_;
+
+  ap_uint<3> iphiSave = iphi_ + phiPlus_;
+  auto secondSave = second_;
+
+  if(zero<kNBits_MemAddrBinned>()[istub_]) {
+     
+    //Need to read the information about the proj in the buffer
+    VMProjection<BARREL> data(projbuffer_.getProjection());
+
+    //FIXME is this valid? Only using range(3,1) instead of full range, zfirst in MatchProcessor.h
+    zbin = data.getZBin().range(3,1); 
```

Comment:
@aryd Would it make sense to define this range within the `VMProjection` class, with meaningful names for the LSB and MSB?